shared/src/main/scala/mlscript/codegen/typescript/TsTypegen.scala

```diff
@@ -431,6 +431,7 @@ final class TsTypegenCodeBuilder {
         // recursive type does not have any other type variables
         // (except itself)
         if (mlType.freeTypeVariables.size === 0) {
+          typeScope.declareTypeAlias(uvName.toString(), List.empty, body)
```

Comment:
Oh, I see this wild `toString` is what resulted in the wildly incorrect example we had before. This is to be absolutely avoided! Use proper typed logic whenever possible, instead of relying on `toString's` behavior.

Reply:
The bug you had is a prime example of that: you assumed the behavior of `toString` to be adequate here, but you were wrong, and you did not realize that because you did not destructure the actual types, which would have shown you the problem.

Reply:
Fixed. By declaring recursive type variables as type aliases, and storing their information as TypeAliasSymbol. So now it looks something like this -

typeVarMapping(α1 -> "b")
typeScope(TypeAliasSymbol("b", ["T"], `the actual type`))

Now when TypeVar is converted it is looked up from mapping. If it's aliased the aliased form i.e. `b<T>` is returned otherwise it's just the name is returned like "a".